src/gen.rs

```diff
@@ -332,16 +327,15 @@ fn gen_fn_type_for_trait(
     };
 
     if let Some((fn_traits, receiver, allowed)) = err {
-        let msg = format!(
+        emit_error!(
+            method.sig.span(),
             "the trait '{}' cannot be auto-implemented for {}, because this method has \
                 {} receiver{}",
             trait_def.ident,
             fn_traits,
             receiver,
             allowed,
         );
-
-        return method.sig.span().err(msg).emit_with_attr_note();
```

Comment:
I think I would prefer to `return TokenStream::new()` here. Most functions are written with the expectation that on error, we return early. So code further down the function might break. I checked this function and it seems fine for this, but I'd still prefer an early return at this point. The three errors above are fine.